mongomock/aggregate.py

```diff
@@ -1364,6 +1365,31 @@ def _handle_sample_stage(in_collection, unused_database, options):
     return shuffled[:size]
 
 
+def _handle_sort_by_count_stage(in_collection, unused_database, options: Union[str, Dict]):
+    key_count = {}
+    sort_by_count = []
+
+    if isinstance(options, Dict) and '$mergeObjects' in options:
+        raise NotImplementedError(
+            "Although '$mergeObjects' is a valid operator for the $sortByCount stage, "
+            'it is currently not implemented in Mongomock.'
+        )
+    field_to_count = options[1:]
+
+    for doc in in_collection:
+        field_value = doc.get(field_to_count)
+        if field_value is None:
+            continue
+        if field_value not in key_count:
+            key_count[field_value] = 0
+        key_count[field_value] += 1
```

Comment:
Hmm... This is probably what [collections.Counter](https://docs.python.org/3/library/collections.html#collections.Counter) is good for. I think you could rewrite all of this as
```python
counter = Counter([doc[group_by] for doc in in_collection if group_by in doc])
return [{"_id": key, "count": count for key, count in counter.most_comon()]
```